### PR TITLE
Make the UI aware of principal-field

### PR DESCRIPTION
**Pull request by cccs-tom**
The Web UI now honours the `http-server.authentication.oauth2.principal-field` config property.
Addresses #7526.

### CODE REVIEW

**Comment by cccs-tom** [comment]
@lukasz-walkiewicz I implemented what we disucssed in the issue comments. However, I was having trouble figuring out how to test this properly, especially the failure paths (where the specified claim is missing or empty). I hesitate to modify some of these public test harnesses that may be used elsewhere in the code. I would appreciate any help / direction you can offer! 😄

**Comment by lukasz-walkiewicz** [comment]
Hi @cccs-tom,

Sorry for the delay. Indeed Hydra does not support custom claims so I think it's not possible to test it this way.
I've created a test using a bit different approach: https://github.com/trinodb/trino/commit/405e29bddeb53aa3c1220d4440cfc94e736375a0. If you like it feel free to squash it with your commit.

File: core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java [hunk 1]
```diff
@@ -150,4 +158,15 @@ private boolean hasValidAudience(Object audience)
...
         return false;
     }
+
+    private boolean hasValidPrincipal(Object principal)
+    {
+        if (principal == null) {
+            return false;
+        }
+        if (principal instanceof String) {
+            return ((String) principal).length() > 0;
```
Comment (lukasz-walkiewicz) [inline core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java:168]: `com.google.common.base.Strings#isNullOrEmpty`

File: core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java [hunk 2]
```diff
@@ -97,9 +99,15 @@ public void filter(ContainerRequestContext request)
...
-            String subject = claims.get().getSubject();
-            setAuthenticatedIdentity(request, Identity.forUser(userMapping.mapUser(subject))
-                    .withPrincipal(new BasicPrincipal(subject))
+            Object principal = claims.get().get(principalField);
+            if (!hasValidPrincipal(principal)) {
+                LOG.debug("Invalid principal field: %s. Expected principal to be non-empty", principalField);
+                sendErrorMessage(request, UNAUTHORIZED, "Unauthorized");
+                return;
+            }
+            String principalName = principal.toString();
```
Comment (lukasz-walkiewicz) [inline core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java:108]: Simple cast? At this point we know that it's an instance of String.class.

File: core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java [hunk 3]
```diff
@@ -150,4 +158,15 @@ private boolean hasValidAudience(Object audience)
         }
         return false;
     }
+
+    private boolean hasValidPrincipal(Object principal)
```
Comment (lukasz-walkiewicz) [inline core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java:162]: nit: `isValidPrincipal`

**Comment by cccs-tom** [comment]
Thank you very much for the test code, @lukasz-walkiewicz, and sorry for the wait. As suggested, I squashed your commit with my other fixes.

File: core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java [hunk 4]
```diff
@@ -150,4 +159,9 @@ private boolean hasValidAudience(Object audience)
         }
         return false;
     }
+
+    private boolean isValidPrincipal(Object principal)
+    {
+        return principal instanceof String && !Strings.isNullOrEmpty((String) principal);
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java:165]: `instanceof` also verifies that `principal != null` so I think you do:
```
        return principal instanceof String && !principal.isEmpty();
```

**Review by kokosing — COMMENTED** [review]
Please provide docs changes too and please squash all your commits.

Thanks. Only a minor comment otherwise looks good.

**Comment by lukasz-walkiewicz** [comment]
Docs have already been prepared in: https://github.com/trinodb/trino/pull/7733

File: core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java (hunk 4 above)
Comment (cccs-tom) [reply core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java]: Of course! I took @lukasz-walkiewicz's suggestion a little too literally! 😄 Fixed and squashed.

**Comment by kokosing** [comment]
Merged as https://github.com/trinodb/trino/commit/9e9b794335808253befbee85d5ca2e1b69c1b432, thanks!

**Pull request closed without merging**